lib/backend/dynamo/dynamodbbk.go

```diff
@@ -838,42 +881,55 @@ func trimPrefix(key string) []byte {
 	return []byte(strings.TrimPrefix(key, keyPrefix))
 }
 
-// create helper creates a new key/value pair in Dynamo with a given expiration
-// depending on mode, either creates, updates or forces create/update
-func (b *Backend) create(ctx context.Context, item backend.Item, mode int) error {
+// create is a helper that writes a key/value pair in Dynamo with a given expiration.
+// Depending on the mode provided, the item will must various conditions met before
+// the item is persisted in the database. On a successful write the revision of the
+// item is returned.
+func (b *Backend) create(ctx context.Context, item backend.Item, mode int) (string, error) {
 	r := record{
 		HashKey:   hashKey,
 		FullPath:  prependPrefix(item.Key),
 		Value:     item.Value,
 		Timestamp: time.Now().UTC().Unix(),
 		ID:        time.Now().UTC().UnixNano(),
+		Revision:  backend.CreateRevision(),
 	}
 	if !item.Expires.IsZero() {
 		r.Expires = aws.Int64(item.Expires.UTC().Unix())
 	}
 	av, err := dynamodbattribute.MarshalMap(r)
 	if err != nil {
-		return trace.Wrap(err)
+		return "", trace.Wrap(err)
 	}
 	input := dynamodb.PutItemInput{
 		Item:      av,
 		TableName: aws.String(b.TableName),
 	}
+
 	switch mode {
 	case modeCreate:
 		input.SetConditionExpression("attribute_not_exists(FullPath)")
 	case modeUpdate:
 		input.SetConditionExpression("attribute_exists(FullPath)")
 	case modePut:
+	case modeConditionalUpdate:
+		input.SetExpressionAttributeNames(map[string]*string{"#v": aws.String("Revision")})
+		input.SetExpressionAttributeValues(map[string]*dynamodb.AttributeValue{":rev": {S: aws.String(item.Revision)}})
+		input.SetConditionExpression("#v = :rev AND attribute_exists(FullPath)")
```

Comment:
Same as in `ConditionalDelete`.

```suggestion
		input.SetExpressionAttributeValues(map[string]*dynamodb.AttributeValue{":rev": {S: aws.String(item.Revision)}})
		input.SetConditionExpression("Revision = :rev AND attribute_exists(FullPath)")
```